mopidy/mpd/protocol/stored_playlists.py

```diff
@@ -145,7 +149,53 @@ def playlistadd(context, name, uri):
 
         ``NAME.m3u`` will be created if it does not exist.
     """
-    raise exceptions.MpdNotImplemented  # TODO
+    uri = context.lookup_playlist_uri_from_name(name)
+    playlist = uri is not None and context.core.playlists.lookup(uri).get()
+    if not playlist:
+        # Create new playlist with this single track
+        lookup_res = context.core.library.lookup(uris=[track_uri]).get()
+        tracks = [a for sl in lookup_res.values() for a in sl]
+        _playlistcreate(context, name, tracks)
+    else:
+        # Add track to existing playlist
+        uri_scheme = urlparse.urlparse(track_uri).scheme
+        lookup_res = context.core.library.lookup(uris=[track_uri]).get()
+        to_add = [a for sl in lookup_res.values() for a in sl]
+        playlist = playlist.replace(tracks=list(playlist.tracks) + to_add)
+        if context.core.playlists.save(playlist).get() is None:
+            playlist_scheme = urlparse.urlparse(playlist.uri).scheme
+            raise exceptions.MpdInvalidTrackForPlaylist(
+                playlist_scheme, uri_scheme)
+
+
+def _playlistcreate(context, name, tracks):
+    """
+    Creates new playlist using backend aprropriate for passed list of tracks
+    """
+    uri_schemes = set([urlparse.urlparse(t.uri).scheme for t in tracks])
+    for scheme in uri_schemes:
+        playlist = context.core.playlists.create(name, scheme).get()
+        if not playlist:
+            # Backend can't create playlists at all
+            logger.warning('%s backend can\'t create playlists', scheme)
+            continue
+        playlist = playlist.replace(tracks=tracks)
+        if context.core.playlists.save(playlist).get() is None:
+            # Falied to save using this backend
```

Comment:
In the situation where my backend can create playlists, but I cannot save the particular tracks in it, I'm going to be left with a unused empty playlist. Can we delete the playlist since we are not using it?